diff --git a/sim/V2/runtime/CMakeLists.txt b/sim/V2/runtime/CMakeLists.txt
index 9c87f8f89..7b4b0c6fb 100644
--- a/sim/V2/runtime/CMakeLists.txt
+++ b/sim/V2/runtime/CMakeLists.txt
@@ -172,8 +172,8 @@ VISIT_INSTALL_TARGETS(simV2runtime_ser)
 
 IF(VISIT_PARALLEL)
     ADD_PARALLEL_LIBRARY(simV2runtime_par ${LIBSIM_RUNTIME_SOURCES} ${LIBSIM_STATIC_SOURCES})
-#    TARGET_LINK_LIBRARIES(simV2runtime_par engine_par vtksys vtkpng vtkmetaio vtkDICOMParser vtkImagingSources vtkglew vtkCommonSystem vtkCommonColor vtkCommonComputationalGeometry vtkFiltersStatistics vtkalglib vtklz4 vtkImagingFourier ${LIBSIM_RUNTIME_VIEWER_LIBS_PAR})
-    TARGET_LINK_LIBRARIES(simV2runtime_par engine_par ${VTK_LIBRARIES} ${LIBSIM_RUNTIME_VIEWER_LIBS_PAR})
+
+    TARGET_LINK_LIBRARIES(simV2runtime_par engine_par ${LIBSIM_RUNTIME_VIEWER_LIBS_PAR})
     IF(NOT APPLE)
         SET_TARGET_PROPERTIES(simV2runtime_par PROPERTIES INSTALL_RPATH "$ORIGIN")
     ENDIF(NOT APPLE)
