diff --git a/avt/VisWindow/CMakeLists.txt b/avt/VisWindow/CMakeLists.txt
index 0065e9ce15..cdcd528281 100644
--- a/avt/VisWindow/CMakeLists.txt
+++ b/avt/VisWindow/CMakeLists.txt
@@ -183,11 +183,17 @@ LINK_DIRECTORIES(${LIBRARY_OUTPUT_DIRECTORY} ${EAVL_LIBRARY_DIR} )
 #********************************* SERIAL ************************************
 ADD_LIBRARY(avtviswindow_ser ${AVTVISWINDOW_SOURCES})
 TARGET_LINK_LIBRARIES(avtviswindow_ser visitcommon avtview avtmath avtplotter_ser visit_vtk lightweight_visit_vtk vtkFiltersModeling vtkImagingCore vtkInteractionStyle ${OBJC_LIBS})
+IF(VISIT_OSPRAY)
+    TARGET_LINK_LIBRARIES(avtviswindow_ser vtkRenderingOSPRay)
+ENDIF(VISIT_OSPRAY)
 VISIT_INSTALL_TARGETS(avtviswindow_ser)
 
 #********************************* PARALLEL **********************************
 IF(VISIT_PARALLEL)
     ADD_PARALLEL_LIBRARY(avtviswindow_par ${AVTVISWINDOW_SOURCES})
     TARGET_LINK_LIBRARIES(avtviswindow_par visitcommon avtview avtmath avtplotter_par visit_vtk lightweight_visit_vtk vtkFiltersModeling vtkImagingCore vtkInteractionStyle ${OBJC_LIBS})
+    IF(VISIT_OSPRAY)
+        TARGET_LINK_LIBRARIES(avtviswindow_par vtkRenderingOSPRay)
+    ENDIF(VISIT_OSPRAY)
     VISIT_INSTALL_TARGETS(avtviswindow_par)
 ENDIF(VISIT_PARALLEL)
